packages/astro/src/internal/fetch-content.ts

```diff
@@ -0,0 +1,22 @@
+export async function fetchContent(importMetaGlobResult: Record<string, () => Promise<any>>, url: string) {
+    if (typeof importMetaGlobResult === 'string') {
+      throw new Error(`[deprecated] "Astro.fetchContent(str)" is now "await Astro.fetchContent(import[dot]meta[dot]glob(str))"`);
+    }
+    const allImporters = [...Object.entries(importMetaGlobResult)];
+    const allImports = await Promise.all(allImporters.map(([spec, imp]) => {
+      return imp().then(mod => {
+        if (!mod.__content) {
+          return;
+        }
+        const urlSpec = new URL(spec, url).pathname.replace(/[\\/\\\\]/, '/');
+        if (!urlSpec.includes('/pages/')) {
+          return mod.__content;
+        }
+        return {
+          ...mod.__content,
+          url: urlSpec.replace(/^.*\/pages\//, '/').replace(/\.md$/, ''),
+        };
+      });
+    }));
+    return allImports.filter(Boolean);
+  }
```

Comment:
This is pretty straightforward! Big fan.

One note: This doesn't restrict to `.md` files like the old implementation. Probably good, but possible performance footgun? Should `.astro` files be supported?

Reply:
It does actually, since only markdown files return a `__content` export. Will add a comment to clarify this.